pkg/helpers/newapp/app/dockerimagelookup.go

```diff
@@ -265,6 +267,46 @@ func (s ImageImportSearcher) Type() string {
 	return "images via the image stream import API"
 }
 
+func selectFromInt(dc interface{}, field string) (int64, error) {
+	in, ok := dc.(map[string]interface{})[field]
+	if ok {
+		return in.(int64), nil
+	}
+	return 0, errors.New("value does not exist")
+}
+
+func selectFromString(dc interface{}, field string) (string, error) {
+	str, ok := dc.(map[string]interface{})[field]
+	if ok {
+		return str.(string), nil
+	}
+	return "", errors.New("value does not exist")
+}
+
+func selectFromArrayOfString(dc interface{}, field string) ([]string, error) {
+	arr, ok := dc.(map[string]interface{})[field]
+	if ok {
+		arx := make([]string, 0, 5)
+		for _, ex := range arr.([]interface{}) {
+			arx = append(arx, ex.(string))
+		}
+		return arx, nil
+	}
+	return nil, errors.New("value does not exist")
+}
+
+func selectFromMapStruct(dc interface{}, field string) (map[string]struct{}, error) {
+	ep, ok := dc.(map[string]interface{})[field]
+	if ok {
+		exs := map[string]struct{}{}
+		for ex := range ep.(map[string]interface{}) {
```

Comment:
When `ep` is `nil`, this panics.  For example, when I used crc 2.17.0+44e15711 and ran `oc new-app --import-mode=PreserveOriginal --search php`.

Reply:
Added TypeOf Guards

Reply:
Not a blocker, since the above example no longer crashes, but using the `reflect` package to avoid panicking seems like a lot.  It would be enough to check the type assertions (per the last paragraph in [the section on type assertions in the language reference](https://go.dev/ref/spec#Type_assertions)).

Reply:
Sure. I just went heavy weight. If you'd like, I can change away from reflect.